cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java

```diff
@@ -70,7 +70,7 @@
    * @return A list of {@link SecureKeyListEntry} for all the keys visible to the user under the given namespace.
    * @throws NamespaceNotFoundException If the specified namespace does not exist.
    * @throws IOException If there was a problem reading from the store.
-   * @throws Exception If we fail to create a filter for the current principal.
+   * @throws UnauthorizedException If the user does not have READ permissions on the namespace.
```

Comment:
No, it does not.
> On Aug 2, 2016, at 5:07 PM, Bhooshan Mogal <notifications@github.com> wrote:
> 
> In cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java <https://github.com/caskdata/cdap/pull/6274#discussion_r73261052>:
> 
> > @@ -70,7 +70,7 @@
> >     * @return A list of {@link SecureKeyListEntry} for all the keys visible to the user under the given namespace.
> >     * @throws NamespaceNotFoundException If the specified namespace does not exist.
> >     * @throws IOException If there was a problem reading from the store.
> > -   * @throws Exception If we fail to create a filter for the current principal.
> > +   * @throws UnauthorizedException If the user does not have READ permissions on the namespace.
> wait, this is wrong, right? This never throws UnauthorizedException. It only returns entities you have some privileges for.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6274/files/b5c602cb63a60e61be6506a3072b89a5be3e40aa..15f0f9e182f2efce31a1667082ccb7a1296e1b8c#r73261052>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk72jS7c2y9jUHLESEEbPhdC3UaqXks5qb9vDgaJpZM4JZI-3>.
>